src/app/public/modules/date-pipe/date.pipe.ts

```diff
@@ -19,49 +15,40 @@ import {
 
 import 'rxjs/add/operator/takeUntil';
 
+import {
+  SkyDateFormatUtility
+} from './date-format-utility';
+
 @Pipe({
   name: 'skyDate',
   pure: false
 })
 export class SkyDatePipe implements OnDestroy, PipeTransform {
-  private get format(): string {
-    return this._format || 'short';
-  }
 
-  private set format(value: string) {
-    if (value && value !== this._format) {
-      this._format = value;
-    }
-  }
+  private defaultFormat = 'short';
 
-  private set locale(value: string) {
-    if (value && value !== this._locale) {
-      this._locale = value;
+  private format: string;
 
-      // Create a new pipe when the locale changes.
-      this.ngDatePipe = new DatePipe(this._locale);
-    }
-  }
+  private defaultLocale: string;
+
+  private locale: string;
 
-  private defaultLocale = 'en-US';
-  private formattedValue: string;
-  private ngDatePipe: DatePipe;
-  private ngUnsubscribe = new Subject<void>();
   private value: Date;
 
-  private _format: string;
-  private _locale: string;
+  private formattedValue: string;
+
+  private ngUnsubscribe = new Subject<void>();
 
   constructor(
     private changeDetector: ChangeDetectorRef,
     private localeProvider: SkyAppLocaleProvider
   ) {
-    this.locale = this.defaultLocale;
+    this.defaultLocale = 'en-US';
```

Comment:
Why is `defaultLocale` set in the constructor, but `defaultFormat` is set up above?

Reply:
Good call. Updated.